### PR TITLE
Issue 13577: JWE consumer support initial drop

### PR DESCRIPTION
**Pull request by ayoho**
For issue #13577

Adds support for server.xml configuration attributes that allow consuming of JWE tokens.

This includes:
- `keyManagementKeyAlias` config attributes added to `<jwtConsumer>` and `<mpJwt>`
- New `CWWKS6056E` message for an error extracting a JWS from a JWE
- Support for consuming a JWE passed as a Bearer token to a protected resource (should also work using the token header/cookie name server config/MP Config attributes since it goes through the same flow)
- Support for consuming a JWE passed to the `Consumer.createJwt(String)` API

This does **not** include:
- Support for the `mp.jwt.decrypt.key.location` MP Config attribute

### CODE REVIEW

**Comment by ayoho** [comment]
#run-libby-bot
#build
#spawn.fullfat.buckets=com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.jwtsso_fat,com.ibm.ws.security.mp.jwt.1.1_fat,com.ibm.ws.security.mp.jwt.1.2_fat

**Comment by ayoho** [comment]
#run-libby-bot
#build
#spawn.fullfat.buckets=com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.jwtsso_fat,com.ibm.ws.security.mp.jwt.1.1_fat,com.ibm.ws.security.mp.jwt.1.2_fat

File: dev/com.ibm.ws.security.jwt/resources/OSGI-INF/metatype/metatype.xml [hunk 1]
```diff
@@ -87,7 +87,8 @@
         <AD id="jwkEndpointUrl" name="%jwkEndpointUrl" description="%jwkEndpointUrl.desc"  required="false"  type="String" />
         <AD id="amrValues" name="%amrValues" description="%amrValues.desc" required="false" type="String" cardinality="400" ibm:beta="true"/>
         <AD id="sslRef" name="%sslRef" description="%sslRef.desc" required="false" type="String" ibmui:uiReference="com.ibm.ws.ssl.repertoire" />
-         <AD id="useSystemPropertiesForHttpClientConnections" name="%useSystemPropertiesForHttpClientConnections" description="%useSystemPropertiesForHttpClientConnections.desc" required="false" type="Boolean" default="false"/>
+        <AD id="useSystemPropertiesForHttpClientConnections" name="%useSystemPropertiesForHttpClientConnections" description="%useSystemPropertiesForHttpClientConnections.desc" required="false" type="Boolean" default="false"/>
+        <AD id="keyManagementKeyAlias" name="internal" description="internal use only" required="false" type="String" ibm:beta="true"/>
```
Comment (teddyjtorres) [inline dev/com.ibm.ws.security.jwt/resources/OSGI-INF/metatype/metatype.xml:91]: TODO: Update to use keyManagementKeyAlias and keyManagementKeyAlias.consumer.desc

File: dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/utils/JweHelper.java [hunk 2]
```diff
@@ -72,6 +98,12 @@ Key getKeyManagementKeyFromTrustStore(JwtConfig jwtConfig) throws KeyStoreExcept
         return JwtUtils.getPublicKey(keyAlias, trustStoreRef);
     }
 
+    Key getJweDecryptionKey(JwtConsumerConfig config) throws KeyStoreException, CertificateException {
```
Comment (teddyjtorres) [inline dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/utils/JweHelper.java:101]: Please confirm if the key value is trace, if so, then tag this method and elsewhere this key is visible in this path with "@Sensitive"

File: dev/com.ibm.ws.security.mp.jwt/resources/OSGI-INF/metatype/metatype.xml [hunk 3]
```diff
@@ -50,6 +50,7 @@
             <Option label="%tokenHeader.Cookie" value="Cookie" />
         </AD>
         <AD id="cookieName" name="%cookieName" description="%cookieName.desc" required="false" type="String" ibm:beta="true" />
+        <AD id="keyManagementKeyAlias" name="internal" description="internal use only" required="false" type="String" ibm:beta="true" />
```
Comment (teddyjtorres) [inline dev/com.ibm.ws.security.mp.jwt/resources/OSGI-INF/metatype/metatype.xml:53]: TODO: Update to use keyManagementKeyAlias and keyManagementKeyAlias.consumer.desc

**Review by teddyjtorres — APPROVED** [review]

**Pull request merged**